gen3/neural-networks/object-detection/yolo-world/main.py

```diff
@@ -0,0 +1,255 @@
+    device_info = dai.DeviceInfo(args.device_ip)
+    device = dai.Device(device_info)
+
+    with dai.Pipeline(device) as pipeline:
+
+        manip = pipeline.create(dai.node.ImageManipV2)
+        manip.setMaxOutputFrameSize(640 * 640 * 3)
+        manip.initialConfig.setOutputSize(
+            640, 640, dai.ImageManipConfigV2.ResizeMode.LETTERBOX
+        )
+
+        if args.video_path is not None:
+            replayNode = pipeline.create(dai.node.ReplayVideo)
+            replayNode.setOutFrameType(dai.ImgFrame.Type.BGR888i)
+            replayNode.setReplayVideoFile(args.video_path)
+
+            replayNode.out.link(manip.inputImage)
+        else:
+            cam = pipeline.create(dai.node.Camera).build(
+                boardSocket=dai.CameraBoardSocket.CAM_A
+            )
+            camOut = cam.requestOutput((640, 640), dai.ImgFrame.Type.RGB888i)
+
+            camOut.link(manip.inputImage)
+
+        model_description = dai.NNModelDescription(
+            model="yolo-world-l", platform="RVC4"
+        )
+        archive_path = dai.getModelFromZoo(model_description, useCached=True)
+        nn_archive = dai.NNArchive(archive_path)
+
+        nn = pipeline.create(dai.node.NeuralNetwork)
+        nn.setNNArchive(nn_archive)
+
+        nn.setBackend("snpe")
+        nn.setBackendProperties({"runtime": "dsp", "performance_profile": "default"})
+
+        detectionParser = pipeline.create(dai.node.DetectionParser)
+        detectionParser.setConfidenceThreshold(args.confidence_threshold)
+        detectionParser.setNumClasses(MAX_NUM_CLASSES)
+        detectionParser.setCoordinateSize(4)
+        detectionParser.setIouThreshold(0.3)
+        detectionParser.setInputImageSize(640, 640)
+        nn.setNumInferenceThreads(1)
+
+        # Linking
+        manip.out.link(nn.inputs["images"])
+        qDet = detectionParser.out.createOutputQueue()
+        # qImg = nn.passthroughs['images'].createOutputQueue()
+        qImg = manip.out.createOutputQueue()
+        nn.out.link(detectionParser.input)
+
+        textInputQueue = nn.inputs["texts"].createInputQueue()
+        nn.inputs["texts"].setReusePreviousMessage(True)
+
+        pipeline.start()
+
+        inputNNData = dai.NNData()
+        inputNNData.addTensor(
+            "texts", text_features, dataType=dai.TensorInfo.DataType.U8F
+        )
+        textInputQueue.send(inputNNData)
+
+        print("Press 'q' to stop")
+        while pipeline.isRunning():  # and replayNode.isRunning():
+            inDet: dai.ImgDetections = qDet.get()
+            inImage: dai.ImgFrame = qImg.get()
+            cvFrame = inImage.getCvFrame()
+            visFrame = draw_detections(cvFrame, inDet.detections, args.class_names)
+            cv2.imshow("Inference result", visFrame)
+            key = cv2.waitKey(1)
+            if key == ord("q"):
+                break
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Yolo World Demo")
+    parser.add_argument(
```

Comment:
Let's unify arguments with basix experiment [here](https://github.com/luxonis/depthai-experiments/blob/gen3/gen3/neural-networks/basic-example/utils/arguments.py) (e.g. this would become [this](https://github.com/luxonis/depthai-experiments/blob/041ad223c5c867ecb9f75725caf88d0c17938feb/gen3/neural-networks/basic-example/utils/arguments.py#L31-L38))